sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -545,11 +522,11 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
           std::string const& packageName,
           std::string const& packageVersion,
           TelemetryOptions options = TelemetryOptions())
-          : m_telemetryId(Azure::Core::Http::_detail::UserAgentGenerator::GenerateUserAgent(
+          : m_telemetryId(Azure::Core::Http::_internal::UserAgentGenerator::GenerateUserAgent(
               packageName,
               packageVersion,
               options.ApplicationId,
-              options.CppStandardVersion))
+              __cplusplus))
```

Comment:
From offline discussion, we want this within the options since `ClientOptions` and eventually `TelemetryOptions` end up getting created, either implicitly or explicitly in user's code whenever they create an instance of a client.

The TelemetryPolicy typically gets created within `.cpp` files within the SDK, and hence the macro might execute in the context of our `CMakeLists.txt` and C++ language version, instead of the user's.